JASP-Engine/JASP/R/reliabilityanalysis.R

```diff
@@ -15,479 +15,456 @@
-
-		# because we supply a correlation matrix and not raw data, we have to add these ourselves
-		relyFit[["total"]][["mean"]] <- mean(dataList[["itemMeans"]])
-		relyFit[["total"]][["sd"]] <- stats::sd(dataList[["itemMeans"]])
-		relyFit[["item.stats"]][["mean"]] <- dataList[["itemMeans"]]
-		relyFit[["item.stats"]][["sd"]] <- dataList[["itemSds"]]
-		relyFit[["nObs"]] <- nObs
-		
-		# calculate confidence interval for chronbach alpha
-		relyFit[["ciAlpha"]] <- .reliabilityAlphaCI(relyFit = relyFit,	ci = options[["confAlphaLevel"]])
-
-		# calculate the greatest lower bound -- only possible for more than 2 variables.
-		if (nVar < 3) {
-
-			relyFit[["glb"]] <- "."
-
-		} else { # try since the glb is error prone icm reverse scaled items. Requires further investigation/ this might be a bug in psych.
-
-			relyFit[["glb"]] <- .quietDuringUnitTest(try(psych::glb(r = dataList[["correlation"]], key = key)[["glb.max"]], silent = TRUE))
-
-		}
-
-		# calculate McDonalds omega
-		omega <- .quietDuringUnitTest(psych::omega(m = dataList[["correlation"]], nfactors = 1, flip = FALSE, plot = FALSE, 
-							  n.iter = 1, n.obs = nObs)[["omega.tot"]])
-
-		# calculate McDonalds omega if item dropped
-		omegaDropped <- NULL
-		if (nVar > 2) {
-			omegaDropped <- numeric(length = nVar)
-			for (i in 1:nVar) {
-
-					omegaDropped[i] <- .quietDuringUnitTest(psych::omega(m = dataList[["correlation"]][-i, -i], 
-													nfactors = 1, n.iter = 1, n.obs = nObs,
-													flip = FALSE, plot = FALSE)[["omega.tot"]])
-			}
-		}
-
-		relyFit[["omega"]] <- omega
-		relyFit[["omegaDropped"]] <- omegaDropped
-
-	}
-
-	return(relyFit)
-
+# Preprocessing functions ----
+.relReadData <- function(dataset, options) {
+  if (!is.null(dataset)) 
+    return(dataset)
+  else 
+    return(.readDataSetToEnd(columns.as.numeric = options$variables, 
+                             columns.as.factor  = NULL, 
+                             exclude.na.listwise = NULL))
+}  
+
+.reliabilityCheckErrors <- function(dataset, options) {
+  
+  # Error check 1: 0 observations for a level of a variable
+  for (variable in options$variables) {
+    
+    column <- dataset[[.v(variable)]]
+    data   <- column[!is.na(column)]
+    levels <- levels(data)
+    
+    for (level in levels) {
+      .hasErrors(
+        dataset              = data[data == level],
+        perform              = "run",
+        type                 = "observations",
+        observations.amount  = "< 1",
+        exitAnalysisIfErrors = TRUE
+      )
+    }
+  }
+  
+  # Error check 2: One or more variables has infinity
+  .hasErrors(dataset = dataset, 
+             perform = perform,
+             type    = "infinity",
+             infinity.target = options$variables,
+             exitAnalysisIfErrors = TRUE)
+} 
+
+# Results functions ----
+.reliabilityComputeResults <- function (jaspResults, dataset, options, ready) {
+  
+  if (!ready) return()
```

Comment:
Why not check the status of `ready` in the table function? Otherwise you have to do it in multiple places.